libexec/revng/revng-check-conventions

```diff
@@ -1,498 +1,727 @@
+        super().__init__("license-check", "__all")
+
+    @staticmethod
+    def skip_test(path: Path) -> bool:
+        return (
+            path.suffix in [".txt", ".md", ".rst", ".dot"]
+            or path.name
+            in [
+                ".gitignore",
+                "pyproject.toml",
+                "clang-format-style-file",
+                ".clang-tidy",
+                "Doxyfile.in",
+            ]
+            or path.stem == "LICENSE"
+        )
+
+    @staticmethod
+    def check_license(content: str) -> bool:
+        return any("See LICENSE.md for details" in line for line in content.splitlines()[:10])
+
+    def process(self, files: Dict[Path, str], args: Dict[str, str]) -> PassResult:
+        missing_license = []
+        for path, content in files.items():
+            if self.skip_test(path):
+                continue
+            if not self.check_license(content):
+                missing_license.append(path)
+        output = generate_error_files("There are files without a license header:", missing_license)
+        return PassResult(output, len(missing_license) == 0)
+
+
+class BashPass(SingleCommandPass):
+    def __init__(self):
+        super().__init__("bash-check", "bash", ("shellcheck",))
+
+    @staticmethod
+    def check_set(content: str) -> bool:
+        return any(
+            "# rcc-ignore: bash-set-flags" in line or "set -euo pipefail" in line
+            for line in content.splitlines()[:10]
+        )
+
+    def process(self, files: Dict[Path, str], args: Dict[str, str]) -> PassResult:
+        shellcheck = super().process(files, args)
+        noset_files = [path for path, content in files.items() if not self.check_set(content)]
+        output = generate_error_files(
+            "There are script files without 'set -euo pipefail':", noset_files
+        )
+        return PassResult.combine(shellcheck, PassResult(output, len(noset_files) == 0))
+
+
+class RevngGeneralPass(Pass):
+    def __init__(self):
+        super().__init__("revng-general", "__all")
+
+    def process(self, files: Dict[Path, str], args) -> PassResult:
+        trailing_whitespace = defaultdict(list)
+        tabs = defaultdict(list)
+        no_end_newline = []
+        for path, content in files.items():
+            lines = content.splitlines()
+            for index, line in enumerate(lines):
+                if line.endswith(" "):
+                    trailing_whitespace[path].append(index)
+                if "\t" in line:
+                    tabs[path].append(index)
+            if content[-1] != "\n":
+                no_end_newline.append(path)
+
+        return PassResult(
+            generate_error_lines("Whitespace at the end of line:", trailing_whitespace)
+            + generate_error_lines("Tabs present:", tabs)
+            + generate_error_files("Files that don't end in a newline:", no_end_newline),
+            empty(trailing_whitespace, tabs, no_end_newline),
+        )
+
+
+class RevngCPass(Pass):
```

Comment:
`RevngC` has a specific meaning in our context, find a better name.

Reply:
Renamed to `CPPCheckPass`